tests/integration/state_machine/test_state_machine.py

```diff
@@ -94,20 +72,30 @@ def test_state_machine_with_successful_collection(injector, mocker):
     )
 
     message, _ = scheduling_utilities.start_mission(mission=mission)
+    assert message.started
     time.sleep(1)
```

Comment:
Can this sleep be removed?

Reply:
It can, but then I will have to make some other logic to assure that the state machine is done running. If I do the assert without waiting, the state machine is not always done running and expected transition log will not match the transition log on the state machine.

Reply:
True, still two threads after all!